pkg/poddisruptionbudget/poddisruptionbudget_test.go

```diff
@@ -0,0 +1,374 @@
+			assert.Nil(t, err)
+			assert.Equal(t, testResourceQuota.Name, builderResult.Object.Name)
+			assert.Equal(t, testResourceQuota.Namespace, builderResult.Object.Namespace)
+		}
+	}
+}
+
+func TestResourceQuotaWithQuotaSpec(t *testing.T) {
+	testCases := []struct {
+		builderNil    bool
+		definitionNil bool
+		expectedError bool
+		errorMessage  string
+	}{
+		{ // Test Case 1 - Happy path, no nils
+			builderNil:    false,
+			definitionNil: false,
+			expectedError: false,
+		},
+		{ // Test Case 2 - Builder is nil
+			builderNil:    true,
+			definitionNil: false,
+			expectedError: true,
+		},
+		{ // Test Case 3 - Definition is nil
+			builderNil:    false,
+			definitionNil: true,
+			expectedError: true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := &Builder{
+			apiClient: clients.GetTestClients(clients.TestClientParams{}).PolicyV1Interface,
+			Definition: &policyv1.PodDisruptionBudget{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testPDB",
+					Namespace: "testNamespace",
+				},
+			},
+		}
+
+		if testCase.builderNil {
+			testBuilder = nil
+		}
+
+		if testCase.definitionNil {
+			testBuilder.Definition = nil
+		}
+
+		oneInt := intstr.FromInt(1)
+		result := testBuilder.WithPDBSpec(policyv1.PodDisruptionBudgetSpec{
+			MinAvailable: &oneInt,
+		})
+
+		if !testCase.expectedError {
+			assert.NotNil(t, result)
+			assert.NotNil(t, testBuilder.Definition)
+			assert.Equal(t, oneInt, *testBuilder.Definition.Spec.MinAvailable)
+		} else {
+			if !testCase.builderNil {
+				assert.NotNil(t, result)
+			} else {
+				assert.Nil(t, result)
+			}
+		}
+	}
+}
+
+func TestPDBCreate(t *testing.T) {
+	testCases := []struct {
+		pdbExistsAlready bool
+	}{
+		{
+			pdbExistsAlready: false,
+		},
+		{
+			pdbExistsAlready: true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		testPDB := generatePDB("testRQ", "testNamespace")
+
+		if testCase.pdbExistsAlready {
+			runtimeObjects = append(runtimeObjects, testPDB)
+		}
+
+		testBuilder := buildTestBuilderWithFakeObjects(runtimeObjects, testPDB.Name, testPDB.Namespace)
+		builderResult, err := testBuilder.Create()
+
+		assert.Nil(t, err)
+		assert.NotNil(t, testBuilder.Object)
+		assert.Equal(t, testPDB.Name, builderResult.Object.Name)
+		assert.Equal(t, testPDB.Namespace, builderResult.Object.Namespace)
+	}
+}
+
+func TestPDBDelete(t *testing.T) {
+	testCases := []struct {
+		pdbExistsAlready bool
+	}{
+		{
+			pdbExistsAlready: false,
+		},
+		{
+			pdbExistsAlready: true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		testPDB := generatePDB("testRQ", "testNamespace")
+
+		if testCase.pdbExistsAlready {
+			runtimeObjects = append(runtimeObjects, testPDB)
+		}
+
+		testBuilder := buildTestBuilderWithFakeObjects(runtimeObjects, testPDB.Name, testPDB.Namespace)
+		err := testBuilder.Delete()
+
+		assert.Nil(t, err)
+	}
+}
+
+func TestResourceQuotaExists(t *testing.T) {
```

Comment:
function needs renamed for this resource

Reply:
Done.